src/iceberg/schema.h

```diff
@@ -54,13 +56,42 @@ class ICEBERG_EXPORT Schema : public StructType {
 
   [[nodiscard]] std::string ToString() const override;
 
+  [[nodiscard]] std::optional<std::reference_wrapper<const SchemaField>> GetFieldByName(
+      std::string_view name, bool case_sensitive) const override;
+
+  [[nodiscard]] std::optional<std::reference_wrapper<const SchemaField>> GetFieldByName(
+      std::string_view name) const;
+
+  [[nodiscard]] std::optional<std::reference_wrapper<const SchemaField>> GetFieldById(
+      int32_t field_id) const override;
+
   friend bool operator==(const Schema& lhs, const Schema& rhs) { return lhs.Equals(rhs); }
 
+  mutable std::unordered_map<int, size_t> id_to_index_;
```

Comment:
```suggestion
  /// Mapping from field id to index of `full_schemafield_`.
  mutable std::unordered_map<int32_t, size_t> id_to_index_;
```

It would be good to add some comment to help readers understand it since it is not obvious.